Release version 0.2.0

archmagos opened the pull request.

archmagos committed:
Release version 0.2.0

archmagos commented:
Just checking that this PR should be pointing to _production_ rather than master @sztupy ?

sztupy commented:
Everything should go to master first

sztupy commented:
The process is te same for every project: we push to master and if it needs hotfixing we thhen cherry-pick to staging&prod

For this thehre's no cherry picking required, simly pushh to master thhen our scripts will take care of moving it to prod

it doesn't matter whihch branch you do theh releases from, just make sure you get the version tagged

sztupy commented on example/client.rb:
```diff
@@ -72,6 +72,7 @@ def create_sender_example
       sender.postal_code = "798983"
       sender.birth_date = "1900-12-31"
       sender.ip = '127.0.0.1'
+      sender.external_id = '2b59def0' # Optional field for custom ID
```
```suggestion
      sender.external_id = 'SENDER-2b59def0' # Optional field for customer ID
```

sztupy commented on example/client.rb:
```diff
@@ -128,6 +136,7 @@ def create_transaction_example
       recipient.requested_currency = 'NGN'
       recipient.payout_method = payout_method
 
+      transaction.external_id = '1f834add' # Optional field for custom ID
```
```suggestion
      transaction.external_id = 'TRANSACTION-1f834add' # Optional field for customer's ID
```

sztupy commented on example/client.rb:
```diff
@@ -110,6 +117,7 @@ def create_transaction_example
       transaction = Bitpesa::Transaction.new
 
       sender = Bitpesa::Sender.new
+      # When adding a sender to transaction, please use either an id or external_id. Providing both will result in a validation error.
```
```suggestion
      # When adding a sender to transaction, please use either an id or external_id. Providing both will result in a validation error.
      # Please see our documentation at https://github.com/bitpesa/api-documentation/blob/master/transaction-flow.md#sender
```

sztupy commented:
Looks good, see some change suggestions. Make sure to rebase it to master first

sztupy committed:
Update example/client.rb

Co-Authored-By: archmagos <fredch87@outlook.com>

sztupy committed:
Update example/client.rb

Co-Authored-By: archmagos <fredch87@outlook.com>

sztupy committed:
Update example/client.rb

Co-Authored-By: archmagos <fredch87@outlook.com>

archmagos commented:
Cheers @sztupy - I misread [these notes](https://github.com/bitpesa/bitpesa-api-specification/wiki/Ruby-SDK-Rubygems-release-process) about releasing from production but obviously that's different from what we're doing here. Those notes suggest I need to submit this gem from the production build, so presumably will need to wait for these changes to make it through to prod before doing the release?

sztupy commented:
Yes, releasing should be done from the production branch, but as this is not a hotfix the production branch should be updated based on the usual procedure (same as for any of our other projects) - master first, then pushing the commits up to production.

sztupy commented:
In any case we should 99% of the cases never push commits to production that are not yet pushed to staging and master first. Master should always contain all of the latest changes.

archmagos commented:
Yep that makes sense. As none of the recent SDK changes have been merged into production yet and are still in master, we will presumably need to wait until then before actually releasing the new gem/package/etc versions?

sztupy commented:
In short:
- you add all changes to master with a PR
- you verify they are okay
- you ask for deployment
- once it's deployed (e.g. merged to prod) - you can publish

archmagos commented:
Ok great. If you're happy with these changes we'll merge and then get it deployed.

archmagos committed:
Clean up client, print returned values

archmagos committed:
Add regenerated specs

archmagos commented:
• Updated ZenTest in gemfile, as previous version prevented builds in Travis
• Replaced the entire spec directory with newly generated specs, as the old ones were out of date and failing
• Updated the return values of example methods in client.rb

Working on fixing the four failing tests here: https://travis-ci.com/bitpesa/bitpesa-sdk-ruby/builds/108800074#L603

archmagos committed:
Remove unnecessary specs, update openapi-ignore

archmagos commented:
Updated specs based on latest SDK gen build all running, and all examples in client.rb are tested and functional. Should be ready to go.

sztupy commented on example/client.rb:
```diff
@@ -86,6 +86,20 @@ def create_sender_example
+    begin
+      opts = { external_id: 'SENDER-2b59def0' }
+      sender = Bitpesa::SendersApi.new
+      p sender.get_senders(opts).object.first
```
make sure you return thhe sender as well

sztupy commented on example/client.rb:
```diff
@@ -72,11 +72,11 @@ def create_sender_example
       sender_request.sender = sender
-      api.post_senders(sender_request)
-      # SenderRequest |
+      p api.post_senders(sender_request)
```
p will return nil, if possible return a useful response

sztupy commented on example/client.rb:
```diff
@@ -148,6 +165,20 @@ def create_transaction_example
+    begin
+      opts = { external_id: 'TRANSACTION-1f834add' }
+      transaction = Bitpesa::TransactionsApi.new
+      p transaction.get_transactions(opts).object.first
```
same here

archmagos committed:
Update returned properties & objects in example
